pkg/awsclient/client.go

```diff
@@ -376,12 +378,21 @@ func NewClient(awsAccessID, awsAccessSecret, token, region string) (Client, erro
 	}, nil
 }
 
-// GetAWSClient generates an awsclient
+// Builder implementations know how to produce a Client.
+type Builder interface {
```

Comment:
Lest you think this name is odd, I originally had it as AWSClientProvider, but my IDE made an excellent point that it would be used as `awsclient.AWSClientProvider`, which "stutters" (it actually used that word).